File: Day-6/majority_element.java
import java.util.*;

public class MajorityElement {
    public static List<Integer> findMajority(int[] arr) {
        int n = arr.length;
        Map<Integer, Integer> freq = new HashMap<>();
        List<Integer> ans = new ArrayList<>();

        for (int num : arr) {
            freq.put(num, freq.getOrDefault(num, 0) + 1);
        }

        int majority = n / 3;
        for (Map.Entry<Integer, Integer> entry : freq.entrySet()) {
            if (entry.getValue() > majority) {
                ans.add(entry.getKey());
            }
        }

        return ans;
    }

    public static void main(String[] args) {
        int[] arr = {1,2,3,1,1,2,2,2};
        List<Integer> result = findMajority(arr);
        System.out.println(result);
    }
}

File: Day-6/majority_element.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> findMajority(vector<int>& arr) {
    int n = arr.size();
    unordered_map<int, int> freq;
    vector<int> ans;

    for (int num : arr) {
        freq[num]++;
    }

    int majority = n / 3;
    for (auto &p : freq) {
        if (p.second > majority) {
            ans.push_back(p.first);
        }
    }

    return ans;
}

int main() {
    vector<int> arr = {1,2,3,1,1,2,2,2};
    vector<int> result = findMajority(arr);

    for (int num : result) cout << num << " ";
    cout << "\n";
    return 0;
}
